### DISCUSSION TITLE
So, Rebar has no connection with Vbios. Is the Vbios update for 30 series graphics cards actually a cover?

### DISCUSSION THREAD

**Discussion by 83532070** [terminatorul/NvStrapsReBar#53] [General]
![mmexport1710335147499](https://github.com/terminatorul/NvStrapsReBar/assets/93812008/9a4a7d11-0577-4cc0-8d60-76b23e5dbf7d)
![mmexport1710335144015](https://github.com/terminatorul/NvStrapsReBar/assets/93812008/e541702b-84e8-45f3-b1ea-0e2bb8b4be3c)

**Comment by terminatorul** [comment]
Your screenshot shows you did not enable ReBAR for your MSI RTX 2070 SUPER. Did you encounter any issues ?

The vBIOS update is NOT a cover. The vBIOS is usually the right way to enable ReBAR on the graphics card. 

But because the NVIDIA vBIOS is digitally signed (and encrypted), users and UEFI modders now have to use the motherboard firmware (UEFI) instead, as a work-around.

**Reply by mirh** [reply]
I thought the vbios restrictions had [been busted](https://www.techpowerup.com/312631/nvidia-bios-signature-lock-broken-vbios-modding-and-crossflash-enabled-by-groundbreaking-new-tools)?

**Reply by terminatorul** [reply]
I can see it only allows cross-flashing a vBIOS from a different card. Or allows modifying vBIOS if the 32-bit checksum matches, but no way to match that, or I dont' understand what to do. It also says later the HMAC (signature) must match ... so the restrictions are really not "busted"

**Reply by mirh** [reply]
Those are the restrictions for Ampere cards and newer. 
Yes, the only serious thing you can do there is cross-flashing, and the hard certificate checks don't really leave much hope for the future. 

But on pascal and turing (and even older) cards, my understanding is that you just have to put up with the checksum integrity checks. 
In a few posts the two generations are still described as not-exactly-the-same-thing, but in the great majority of them Veii seemed to argue it was now just a matter of not having good editing tools.

**Reply by terminatorul** [reply]
I could not thoroughly go through the conversation, as their terms are not familiar to me, and the conversation does not explain much.

But if it really works to modify the vBIOS for Pascal and Turing cards, it will help so much all those people that can not update their system firmware (UEFI image) and are raising tickets here for help. But then I also know the NVIDIA vBIOS is encrypted ... at least for desktops, so you see I am not sure what I should believe any more.

Can you add the right link here so I can read the messages ?

**Reply by mirh** [reply]
https://www.techpowerup.com/forums/threads/omg-vflash-fully-patched-nvflash-from-x-to-ada-lovelace-v5-780.312601/page-2
https://www.overclock.net/threads/official-nvflash-with-certificate-checks-bypassed-for-gtx-950-960-970-980-980ti-titan-x.1521334/page-52
https://www.overclock.net/threads/nvidia-security-chip-on-modern-nvidia-video-cards-cracked-modified-and-custom-bioses-allowed-soon.1807471/#post-29228420
Now, to be absolutely fair this is always only one single source, but he is after all the one that gave us these tools to begin with.
